src/NuGet.Services.SearchService/App_Start/WebApiConfig.cs

```diff
@@ -33,7 +32,14 @@ public static class WebApiConfig
     {
         private const string ControllerSuffix = "Controller";
         private const string ConfigurationSectionName = "SearchService";
-        private static readonly TimeSpan KeyVaultSecretCachingTimeout = TimeSpan.FromDays(1);
+
+        /// <summary>
+        /// We use <see cref="int.MaxValue"/> number of seconds for the caching timeout to essentially cache the
+        /// KeyVault secrets for the entire life of the process. This duration is over 68 years so... close enough. 
+        /// This is necessary because the KeyVault secret injector waits for an asynchronous operation to complete
+        /// in a synchronous context. This can cause deadlocks in ASP.NET when this occurs in request context.
+        /// </summary>
+        private static readonly TimeSpan KeyVaultSecretCachingTimeout = TimeSpan.FromSeconds(int.MaxValue);
```

Comment:
If you don't intend to refresh configuration in the process and load it during startup only, why use CachingSecretReader at all? You can use SecretReader directly. I don't have a strong opinion. Just sayin..

Reply:
I haven't dug through the various implementations we have fully -- maybe I am missing something. I just looked through ServerCommon. The `KeyVaultReader` (couldn't find "SecretReader") doesn't cache at all so each config activation via DI would be hitting KeyVault if we used that.

I considered resolving each of the strings directly then putting the resolved settings object in the DI container as a singleton but this would require a fair bit of custom code or the construction of two DI service providers (one for the first resolution one to hold a singleton). I favored this approach I implemented because I was hoping we could fix our KeyVault adapter to Microsoft.Extensions.Configuration and get auto-secret rotation in all the places (just based off of an arbitrary cache lifetime). Then we could just reduce the cache duration here and be done with it.

Happy to take another approach I just don't see the simpler path.